src/main/java/de/tum/in/www1/artemis/config/Constants.java

```diff
@@ -257,6 +257,18 @@ public final class Constants {
      */
     public static final int SIZE_OF_UNSIGNED_TINYINT = 255;
 
+    /**
+     * The regex matches range of decimal numbers
+     * For example: -10.05--1.5
+     * Group 1: -10.05 (Lower Bound)
+     * Group 2: 10.05
+     * Group 3: .05
+     * Group 4: -1.5 (Upper Bound)
+     * Group 5: 1.5
+     * Group 6: .5
+     */
+    public static final String SHORT_ANSWER_NUMBER_SPOT_TYPE_SOLUTION_REGEX = "^(-?(\\d+(\\.\\d+)?|\\.\\d+))\\s*-\\s*(-?(\\d+(\\.\\d+)?|\\.\\d+))$";
```

Comment:
I would not make it that complicated tbh. Just always let the user input a range in a format that is easily parsable without that complicated regex. And always make it inclusive in both ends: For example:
- Only three is allowed `[3|3]`
- Only 4.3 is allowed `[4.3|4.3]`
- 2 to 5.7 is allowed `[2|5.7]`



Simple then remove the `[]` from both ends of the string and split by `|`.

Reply:
I adjusted the implementation based on your suggestion. Thanks!